### PR TITLE
Fix: no-unused-expressions rule incorrectly flagging yield (fixes #1672)

### LINKED ISSUE #1672: bug: yield is not an unused expression
ESLint 0.12.0
with .eslintrc
```js
{
  "ecmaFeatures" : {
    "generators": true
  }
}
```
If `no-unused-expressions` is enabled, the following is considered an error

```js
function *generator() {
    yield "44";
    yield "55";
}
```

From what the newbie me sees, it seems it would be just a matter of modifying line 63 of lib/rules/no-unused-expressions.js to

```js
!/^(?:Assignment|Call|New|Update|Yield)Expression$/.test(type) &&
```

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/7859591-bug-yield-is-not-an-unused-expression?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by remitbri**

### CODE REVIEW

File: docs/rules/no-unused-expressions.md [hunk 1]
```diff
@@ -59,3 +59,10 @@ delete a.b
 ```js
 void a
 ```
+
+
+Additionally in ECMAScript 6, when using a generator, this is also valid:
+
+```js
+yield 0;
```
Comment (nzakas) [inline docs/rules/no-unused-expressions.md]: This isn't necessary. Flagging `yield` was a bug, there's no need to call this out in the docs.

**Comment by nzakas** [comment]
Thanks, I don't think the changes to the docs are necessary, so if you can remove that, this looks good.

Also, please take a moment to sign out CLA: http://eslint.org/cla. This basically says you give us permission to use the code you submit.

Thanks!

**Comment by remitbri** [comment]
The docs are back to unchanged and the the CLA is signed! Thanks.

**Comment by nzakas** [comment]
Awesome, looks good. Thanks!

**Pull request merged**